### DISCUSSION TITLE
Taproot &  Schnorr Signatures

### DISCUSSION THREAD

**Discussion by Conza88** [bisq-network/bisq#5186] [Q&A]
Had a read of [What Is Taproot and How Will It Benefit Bitcoin?](https://river.com/learn/what-is-taproot/) which I thought was pretty good summary.

My questions are—how does this effect Bisq? Is there anything that would need to be implemented/upgraded? Will it be another big task i.e. Segwit? Does this allow for any of Bisq's 'issues' (e.g. presence of Burningman acct etc.) to be solved?

**Comment by ghost** [comment]
At the very least, there is a substantial mining fee savings, as MuSig results in one public key and one signature, while multisig has 2 public keys and 2 signatures. Further, point time locked contracts can be used to replace the hashed time locked contracts of fallback transactions, which are again [a little bit] smaller.

There is also a substantial privacy improvement, as MuSig & PTLCs shares the anonset of singlesig transactions.

There are more future magic improvements too, like using DLC oracles for trust less price inclusion.

**Reply by Conza88** [reply]
That doesn't come automatically to Bisq, right? Would need an update?

**Reply by ghost** [reply]
It's a major change, like really really big.
Maybe a naive implementation can be done rather easily, but to utilize the full power of Schnorr requires both research and development.

**Reply by Conza88** [reply]
Gotcha. Dang. Would be cool if Bisq is one of the first with it implemented.

**Comment by Conza88** [comment]
This was great: [https://www.nasdaq.com/articles/how-bitcoins-taproot-upgrade-will-improve-technology-across-bitcoins-software-stack-2021](https://www.nasdaq.com/articles/how-bitcoins-taproot-upgrade-will-improve-technology-across-bitcoins-software-stack-2021)
